File: 백준/Silver I/1931. 회의실 배정/회의실 배정.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.*;

public class Main {
    static BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    static StringTokenizer st;
    static StringBuilder sb = new StringBuilder();

    public static void main(String[] args) throws IOException {

        int N = Integer.parseInt(br.readLine());

        int[][] lst = new int[N][2];

        for (int i = 0; i < N; i++) {
            st = new StringTokenizer(br.readLine());
            lst[i][0] = Integer.parseInt(st.nextToken());
            lst[i][1] = Integer.parseInt(st.nextToken());
        } // 입력 완료

        Arrays.sort(lst, Comparator.comparingInt((int[] o) -> o[1]).thenComparingInt(o -> o[0]));

        int endTime = -1;
        int answer = 0;
        for (int[] time:lst) {
            if (time[0] >= endTime) {
                answer += 1;
                endTime = time[1];
            }
        }

        System.out.println(answer);

    } // end of main

} // end of class

File: 백준/Silver I/1931. 회의실 배정/회의실 배정.py
"""
완탐?
n이 100_000 2^100_000

그리디
가장 빨리 끝나는 회의를 우선으로 잡자
"""
import sys

input = lambda: sys.stdin.readline().strip()

N = int(input())
lst = []
for _ in range(N):
    a, b = map(int, input().split())
    lst.append([b, a])

lst = sorted(lst)

end_time = -1
answer = 0
for e, s in lst:
    if s >= end_time:
        answer += 1
        end_time = e

print(answer)